File: src/my_cpp_pkg/src/robot_news_station.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"
 
class RobotNewsStation : public rclcpp::Node // MODIFY NAME
{
public:
    RobotNewsStation() : Node("RobotNewsStation") // MODIFY NAME
    {
        publisher_ = this->create_publisher<example_interfaces::msg::String>("robot_news_cpp", 10);
        timer_ = this->create_wall_timer(
            std::chrono::milliseconds(500),
            std::bind(&RobotNewsStation::timerCallback, this)
        );
        RCLCPP_INFO(this->get_logger(), "RObotNews Station CPP created");
    }
 
private:
    int counter = 1;

    rclcpp::TimerBase::SharedPtr timer_;

    void timerCallback(){
        publishNews();
    }
    void publishNews(){
        auto msg = example_interfaces::msg::String();
        msg.data = std::string("Hello ") + std::to_string(counter);
        counter++;
        publisher_->publish(msg);
    }
    rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher_;
};
 
int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<RobotNewsStation>(); // MODIFY NAME
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/robot_news_station.py
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String
 
class RobotNewsStationNode(Node): 
    '''
    a node that publishes hello to the topic "robot_news" once every 30 seconds
    '''
    
    def __init__(self):
        super().__init__("robot_news_station")
        self.publisher_ = self.create_publisher(String, "robot_news", 10) # data type, topic name, queue size
        self.ctr = 1
        self.create_timer(0.5, self.timer_callback)
        self.get_logger().info("RobotNewsStation has started")
    
    def publish_news(self):
        msg = String()
        msg.data = "Hello " + str(self.ctr)
        self.publisher_.publish(msg)
        self.ctr += 1
        
    def timer_callback(self):
        self.publish_news()
 
def main(args=None):
    rclpy.init(args=args)
    node = RobotNewsStationNode()
    rclpy.spin(node)
    rclpy.shutdown()
 
 
if __name__ == "__main__":
    main()